src/storage-controller/src/lib.rs

```diff
@@ -2140,6 +2105,12 @@ where
             .collect();
         self.append_shard_mappings(shards_to_update.into_iter(), -1);
 
+        for id in pending_collection_drops {
+            self.collections
+                .remove(&id)
+                .expect("list populated after checking that self.collections contains it");
+        }
```

Comment:
This comment must be referring back to a very old design goal of having the storage controller permanently bind a global id with a description. This is described in Frank's architecture design doc. For that to work durable state is required but we don't have that so that comment is.. not correct.

I don't know why exports are not cleaned up, that looks like a bug to me because entries will certainly be removed from there once environment restarts and adapter never mentions the dropped exports again (because it has also forgotten about them).

I will delete that comment altogether